core/contracts/oracle/implementation/EncryptedStore.sol

```diff
@@ -0,0 +1,76 @@
+pragma solidity ^0.5.0;
+
+/**
+ * @title Contract for storing keyed encrypted messages via the EVM. Only authorized senders can store
+ * messages for a given address' store.
+ * @dev This contract uses topic hashes as keys and can store a single arbitrary encrypted message per topic at any
+ * given time.
+ * @dev There's technically nothing that requires the topics hashed or for the messages to be encrypted.
+ * @dev This contract is built for the following specific use case:
+ * - The sender knows the topics ahead of time. This can either be communicated elsewhere or be implicit.
+ * - Only one message per topic is stored at any given time.
+ */
+contract EncryptedStore {
+    struct TopicData {
+        // The encrypted message.
+        bytes message;
+    }
+
+    struct Store {
+        // This maps from a hash to the data for this topic.
+        // Note: the hash is a hash of the "subject" or "topic" of the message.
+        mapping(bytes32 => TopicData) topics;
+        // This contains the set of all authorized senders for this store.
+        mapping(address => bool) authorizedSenders;
```

Comment:
Do we still need authorized senders? I thought the idea behind this change was to make this a place for a user to store their own data rather than send messages to other users?

Reply:
No we do not, I realized this right after I made the initial push.